src/main/java/com/blackhearth/blockchain/node/BlockChainNodeException.java

```diff
@@ -1,6 +1,6 @@
 package com.blackhearth.blockchain.node;
 
-public class BlockChainNodeException extends Throwable {
+public class BlockChainNodeException extends RuntimeException {
```

Comment:
Jeśli zmieniasz z Throwable na RuntimeException to pozbądź się wszystkich throws ;).
Nie jestem pewny czy chcemy mieć RuntimeException tutaj. Przeanalizuj to Krzysiu.